src/code.sh

```
@@ -81,150 +44,325 @@ _trim_fastq_endings () {
-       if [[ ! "${R2_test}" =~ "${R1_test[$i]}" ]]; then 
-              echo "Each R1 FASTQ does not appear to have a matching R2 FASTQ"
-              echo "${R2_test} ${R1_test[$i]}"
-              exit 1 
+_sense_check_fastq_arrays() {
+       : '''
+       Running tests to sense-check R1/R2 arrays and file prefixes
+       '''
+       # Check that there are the same number of R1s as R2s
+       readarray -t R1 < <(find . -maxdepth 1 -name '*_R1_*' -printf '%f\n')
+       readarray -t R2 < <(find . -maxdepth 1 -name '*_R2_*' -printf '%f\n')
+       R2=($(ls *_R2_*))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)